robot_ws/src/voice_interaction_robot/launch/voice_interaction.launch.py

```diff
@@ -0,0 +1,159 @@
+"""
+ Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ software and associated documentation files (the "Software"), to deal in the Software
+ without restriction, including without limitation the rights to use, copy, modify,
+ merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ permit persons to whom the Software is furnished to do so.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import os
+import sys
+
+import yaml
+import launch
+import launch_ros.actions
+from ament_index_python.packages import get_package_share_directory
+
+
+def generate_launch_description():
+    default_lex_config = os.path.join(get_package_share_directory('voice_interaction_robot'), 'config', 'lex_config.yaml')
+
+    with open(default_lex_config, 'r') as f:
+      config_text = f.read()
+    config_yaml = yaml.safe_load(config_text)
+    default_aws_region = config_yaml['lex_node']['ros__parameters']['aws_client_configuration']['region']
+    default_lex_user_id = config_yaml['lex_node']['ros__parameters']['lex_configuration']['user_id']
+
+    ld = launch.LaunchDescription([
+        launch.actions.DeclareLaunchArgument(
+            name='aws_region',
+            default_value=os.environ.get('ROS_AWS_REGION', default_aws_region),
+            description='AWS region override, defaults to config .yaml if unset'
```

Comment:
This launch argument is declared but never used. As is, the elx node just uses the default region in its config file no matter what is set here.

Reply:
Ah true. Fixed, and submitted PR for lex_node to allow setting region and user_id via param https://github.com/aws-robotics/lex-ros2/pull/8/.